CorsixTH/Lua/entities/humanoids/staff.lua

```diff
@@ -154,37 +154,43 @@ function Staff:tick()
   self:updateSpeed()
 end
 
-function Staff:checkIfWaitedTooLong()
+function Staff:checkIfWaitedTooLongForRaise()
   if self.quitting_in then
     self.quitting_in = self.quitting_in - 1
-    if self.quitting_in < 0 then
+
+    local is_waiting_time_is_up = self.quitting_in < 0
+    if is_waiting_time_is_up then
       local rise_windows = self.world.ui:getWindows(UIStaffRise)
       local staff_rise_window = nil
+      local rise_amount = self.profile:getRiseAmount()
+      self.quitting_in = nil
+      self.hospital:removeMessage(self)
```

Comment:
Should be processed at the very end (you can also remove it from `Staff:fire()` if you do this, I think.

Reply:
> Should be processed at the very end (you can also remove it from `Staff:fire()` if you do this, I think. Edit: Ah no you can't remove it from `Staff:fire()` as you can fire in multiple ways.

Yeap.